lib/User_Proxy.php

```diff
@@ -224,11 +226,22 @@ public function getDisplayName($uid) {
 
 	/**
 	 * checks whether the user is allowed to change his avatar in ownCloud
+	 *
 	 * @param string $uid the ownCloud user name
 	 * @return bool either the user can or cannot
+	 * @throws \OutOfBoundsException
+	 * @throws \InvalidArgumentException
+	 * @throws \BadMethodCallException
+	 * @throws ServerNotAvailableException
+	 * @throws DoesNotExistOnLDAPException
 	 */
 	public function canChangeAvatar($uid) {
-		return $this->handleRequest($uid, 'canChangeAvatar', [$uid]);
```

Comment:
I'll need confirmation if this is the way we want to handle this.

With multiple servers, the `handleRequest` includes logic to access directly to the server where the user is. If we have servers 1, 2 and 3, and we know the user is in server 3, the `handleRequest` will access directly to the server 3 without accessing to 1 and 2.
In that scenario, this code will still hit server 1 and 2 always.

Reply:
I've tried to restore this by returning null as well as a random number if the user doesn't exists, but it still doesn't work. It seems there are more things to take into account